block.php

```diff
@@ -27,7 +27,9 @@
 
 $action = optional_param('action', 'view', PARAM_ALPHANUMEXT);
 
-$PAGE->set_context(context_system::instance());
+require_login();
```

Comment:
I think this not required as handled by a controller later in the code.  The same with` $PAGE->set_context(\context_system::instance());` Not obvious...

Reply:
ok. Looks like this is for CI to be happy. Should be ok to leave if doesn't trigger any errors/warnings.